crates/router/src/lib.rs

```diff
@@ -73,16 +72,34 @@ impl Guest for Component {
     }
 }
 
+fn find_matching_route<'a>(routes: &'a [(&'a str, String)], path: &str) -> Option<RouteMatch> {
```

Comment:
Taking a look at the code again, I see we're only looking at the routes in the manifest to find the matching route. We're also collecting those routes into a `Vec` for no reason. 

It might be better for this function to just take a `& AppManifest` and encapsulate the entire logic around fetching routes from the manifest. Thoughts?

Reply:
I tried it and I'm definitely coming round to it!  Thanks for the suggestion!  I've left it as a separate commit in case you aren't keen though.

Reply:
Incidentally, as far as I can tell, we still need a Vec to establish a long-enough lifetime for some borrows (because `Router::build` expects borrowed values).  However, with this change, we can use the "top of function" Vec that is now moved into `find_matching_route`, and no longer need to have a second Vec as in the first iteration of the code.